dotCMS/src/main/java/com/dotmarketing/factories/PublishFactory.java

```diff
@@ -165,6 +188,74 @@ public static boolean publishAsset(Inode webAsset, User user, boolean respectFro
 		return publishAsset(webAsset,user,respectFrontendRoles, true);
 
 	}
+
+	/**
+	 * This method publish a given and asset and its related assets
+	 * if a user is passed the method will check permissions to publish
+	 * if the user doesn't have permission to publish one of the related assets
+	 * then that one will be skipped
+	 * @param folder
+	 * @param user
+	 * @param respectFrontendRoles
+	 * @param isNewVersion  - if passed false then the webasset's mod user and mod date will NOT be altered. @see {@link ContentletAPI#checkinWithoutVersioning(Contentlet, java.util.Map, List, List, User, boolean)}checkinWithoutVersioning.
+	 * @return
+	 * @throws WebAssetException
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 */
+	@SuppressWarnings("unchecked")
+	public static boolean publishAsset(Folder folder, User user, boolean respectFrontendRoles,
```

Comment:
I think this method should be in some PublishAPI rather than Factory, since Factories should not talk with API.
However I may understand sometimes we do not have other choice, think about it to see if it is space to refactor